### DISCUSSION TITLE
Has this kind of optimazation been done?

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/roslyn#58739] [Q&A]
```
public void _Function (A arg1 = default, B arg2 = default, C arg3 = default){}
```

gets optimized :

```
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private void _Function_ (A arg1 = default, B arg2 = default, C arg3 = default){}
public void _Function(A arg1) =>_Function_ (arg1);
public void _Function(B arg2) =>_Function_ (arg2:arg2);
public void _Function(C arg3) =>_Function_ (arg3:arg3);
...Continues
```

**Comment by huoyaoyuan** [comment]
This changes API surface, so it's very unlikely to be done in this way.
This isn't an optimization at all. If the JIT can optimize based on the default value of arguments (for example if they are integer 0, and the function body is inline-able), it will do without the help of single-argument functions.
